RecoLocalCalo/EcalDeadChannelRecoveryAlgos/src/EcalDeadChannelRecoveryBDTG.cc

```diff
@@ -0,0 +1,149 @@
+//  Original Authors:  S. Taroni, N. Marinelli
+//  University of Notre Dame - US
+//  Created:
+//
+//
+//
+
+#include "RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryBDTG.h"
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h"  // can I use a egammatools here?
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include <iostream>
+
+#include <iostream>
+#include <ostream>
+#include <string>
+#include <fstream>
+
+template <>
+void EcalDeadChannelRecoveryBDTG<EBDetId>::addVariables(TMVA::Reader *reader) {
+  for (int i = 0; i < 9; ++i) {
+    if (i == 4)
+      continue;
+    reader->AddVariable("E" + std::to_string(i + 1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx_.rEn[i]));
+  }
+  reader->AddVariable("E1+E2+E3+E4+E6+E7+E8+E9", &(mx_.sumE8));
+  for (int i = 0; i < 9; ++i)
+    reader->AddVariable("iEta" + std::to_string(i + 1), &(mx_.ieta[i]));
+  for (int i = 0; i < 9; ++i)
+    reader->AddVariable("iPhi" + std::to_string(i + 1), &(mx_.iphi[i]));
+}
+template <>
+void EcalDeadChannelRecoveryBDTG<EBDetId>::loadFile() {
+  readerNoCrack = new TMVA::Reader("!Color:!Silent");
+  readerCrack = new TMVA::Reader("!Color:!Silent");
+
+  this->addVariables(readerNoCrack);
+  this->addVariables(readerCrack);
```

Comment:
I can certainly remove the this-> but .. again, is it of an harm ? Just for my own education

Reply:
the harm is obscured readability of the code.
Since ```this->XYZ``` is typically used for disambiguation with another XYZ, it takes an effort to think or check why the extra code was added here.

Reply:
this is fixed. All this-> have been removed in the latest push